### PR TITLE
Create prometheus datasource flow

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
- Adds create prometheus datasource from UI flow
- Uses Application navigateToApp in order to navigate between different plugins
- Shows different linking cards in the data connections page based on the datasource type

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/datasources/components/new/review_s3_datasource_configuration.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ import {
   EuiFlexItem,
   EuiButton,
 } from '@elastic/eui';
+import { Role } from 'common/types/data_connections';
```
Comment (derek-ho) [inline public/components/datasources/components/new/review_s3_datasource_configuration.tsx:16]: Missed this in the previous PR

File: opensearch_dashboards.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
     "savedObjects",
     "uiActions",
     "urlForwarding",
-    "visualizations"
+    "visualizations",
+    "management"
```
Comment (derek-ho) [inline opensearch_dashboards.json:20]: This is to add redirection to Management datasources on the opensearch card - not sure if we should add this here or in separate PR since it seems like that may or may not happen

File: public/components/datasources/components/new/configure_prometheus_datasource.tsx [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    switch (selectedAuthOption) {
+      case 'basic':
+        return (
+          <>
+            <EuiFormRow label="Username">
+              <EuiFieldText
+                placeholder={'Username placeholder'}
+                value={username}
+                onChange={(e) => setUsername(e.target.value)}
+                onBlur={(e) => setUsernameForRequest(e.target.value)}
```
Comment (derek-ho) [inline public/components/datasources/components/new/configure_prometheus_datasource.tsx:93]: @joshuali925 @ps48 was trying to figure a way around all this boiler plate code - issue is that when form is in a child component, re-rendering is causing it to lose focus when input changes. Am working around it with onBlur and temp var. in child component, which defaults to parent variable on load - is there a better way to do this?

File: public/components/datasources/components/new/new_datasource_card_view.tsx [hunk 4]
```diff
@@ -16,33 +16,51 @@ import {
...
+    {
+      name: 'OPENSEARCH',
+      displayName: 'OpenSearch',
+      description: 'Connect to self managed OpenSearch clusters',
+      displayIcon: <EuiIcon type="logoOpenSearch" size="xl" />,
+      onClick: () =>
+        application!.navigateToApp(MANAGEMENT_APP_ID, {
+          path: 'opensearch-dashboards/dataSources',
+        }),
+    },
```
Comment (derek-ho) [inline public/components/datasources/components/new/new_datasource_card_view.tsx:47]: May want to just remove this card due to scope creep - wdyt?

File: public/components/datasources/components/manage/data_connection.tsx [hunk 5]
```diff
@@ -103,6 +107,48 @@ export const DataConnection = (props: any) => {
...
+    switch (datasourceDetails.connector) {
+      case 'S3GLUE':
+        return (
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <EuiCard
+                icon={<EuiIcon size="xxl" type="discoverApp" />}
+                title={'Query data'}
+                description="Query your data in Data Explorer or Observability Logs."
+                onClick={() => application!.navigateToApp(observabilityLogsID)}
```
Comment (derek-ho) [inline public/components/datasources/components/manage/data_connection.tsx:121]: Need another set of eyes here that this is the right way to navigate between apps

File: public/components/datasources/components/manage/data_connection.tsx [hunk 6]
```diff
@@ -103,6 +108,99 @@ export const DataConnection = (props: any) => {
...
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiCard
+                icon={<EuiIcon size="xxl" type="bolt" />}
+                title={'Accelerate performance'}
+                description="Accelerate performance through OpenSearch indexing."
+                onClick={() => {}}
```
Comment (derek-ho) [inline public/components/datasources/components/manage/data_connection.tsx:130]: @sumukhswamy @ps48 still not sure what this card is supposed to do, since acceleration is no longer another page?

File: public/components/datasources/components/manage/data_connection.tsx [hunk 7]
```diff
@@ -103,6 +108,99 @@ export const DataConnection = (props: any) => {
...
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiCard
+                icon={<EuiIcon size="xxl" type="database" />}
+                title={'Tables'}
+                description="Manually Define Tables"
+                onClick={() => application!.navigateToApp('opensearch-query-workbench')}
```
Comment (derek-ho) [inline public/components/datasources/components/manage/data_connection.tsx:138]: Will keep it like this for now, but I think ideally there should be additional path needed for workbench to correctly display the right datasource?

File: public/components/datasources/components/new/new_datasource_card_view.tsx [hunk 8]
```diff
@@ -16,33 +16,37 @@ import {
...
+  onClick: () => void;
 }
 
-const Datasources: DatasourceCard[] = [
-  {
-    name: 'OPENSEARCH',
-    displayName: 'OpenSearch',
-    description: 'Connect to self managed OpenSearch clusters',
-    displayIcon: <EuiIcon type="logoOpenSearch" size="xl" />,
-  },
```
Comment (derek-ho) [inline public/components/datasources/components/new/new_datasource_card_view.tsx:34]: Removed opensearch for now - if we can add it back later in the right way will do so - this specific card needs to not be shown if datasources is not there

File: public/framework/core_refs.ts [hunk 9]
```diff
@@ -3,7 +3,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { ChromeStart, HttpStart, IToasts } from '../../../../src/core/public';
+import {
+  ApplicationStart,
+  ChromeStart,
+  HttpStart,
+  IToasts,
+  IUiSettingsClient,
```
Comment (ps48) [inline public/framework/core_refs.ts:11]: are we using `IUiSettingsClient` anywhere?

File: public/components/datasources/components/new/configure_prometheus_datasource.tsx (hunk 3 above)
Comment (ps48) [reply public/components/datasources/components/new/configure_prometheus_datasource.tsx:93]: What is the cause of re-rendering? Suspecting: Should the switch statement be in a useEffect()?

File: public/components/datasources/components/manage/connection_configuration.tsx [hunk 10]
```diff
@@ -127,7 +126,7 @@ export const ConnectionConfiguration = (props: ConnectionConfigurationProps) =>
           </EuiFormRow>
           <EuiFormRow label="Password">
             <EuiFieldPassword
-              type={dual ? 'dual' : undefined}
+              type={'dual'}
```
Comment (ps48) [inline public/components/datasources/components/manage/connection_configuration.tsx:129]: Minor: We can just keep this`type="dual"`

File: public/components/datasources/components/manage/data_connection.tsx (hunk 5 above)
Comment (ps48) [reply public/components/datasources/components/manage/data_connection.tsx:122]: IMO This is good!

File: public/components/datasources/components/manage/data_connection.tsx (hunk 6 above)
Comment (ps48) [reply public/components/datasources/components/manage/data_connection.tsx:130]: This will redirect to query workbench with accelerate table flyout open. Can keep it blank for now.

File: public/components/datasources/components/manage/data_connection.tsx (hunk 7 above)
Comment (ps48) [reply public/components/datasources/components/manage/data_connection.tsx:138]: @sumukhswamy  Can help you out here. We need to pass in `current data source` here in URL.

File: public/components/datasources/components/new/configure_prometheus_datasource.tsx (hunk 3 above)
Comment (derek-ho) [reply public/components/datasources/components/new/configure_prometheus_datasource.tsx:93]: What I am thinking - I am passing down the value for that field (for example to persist across navigating to different part of the form and coming back). However, when the child component's onChange is triggered, this updates the parent's value of the variable. This then causes all the children to re-render, and the field loses focus, since it isn't focused by default. The way I am working around this for now/in this PR is to only trigger the parent's update `setValue` hook `onBlur` (i.e. when the user stops typing into the field). However, this is causing a lot of boiler plate with needing another temporary value in the child component. Is there a cleaner way around this?

File: public/components/datasources/components/new/configure_datasource.tsx [hunk 11]
```diff
@@ -73,13 +94,37 @@ export function Configure(props: ConfigureDatasourceProps) {
...
+            currentPassword={password}
+            setPasswordForRequest={setPassword}
+            selectedQueryPermissionRoles={selectedQueryPermissionRoles}
+            setSelectedQueryPermissionRoles={setSelectedQueryPermissionRoles}
+            currentAccessKey={accessKey}
+            currentSecretKey={secretKey}
+            setAccessKeyForRequest={setAccessKey}
+            setSecretKeyForRequest={setSecretKey}
+            currentRegion={region}
+            setRegionForRequest={setRegion}
```
Comment (ps48) [inline public/components/datasources/components/new/configure_datasource.tsx:125]: Question: Is there a way to reduce the number of props here?

Comment (derek-ho) [reply public/components/datasources/components/new/configure_datasource.tsx:125]: This is side effect of this issue: https://github.com/opensearch-project/dashboards-observability/pull/1054#discussion_r1340408705, if there is a cleaner way to do it I am all for it, but I couldn't find one with quick search

**Review by ps48 — APPROVED** [review]
Added minor comments

File: public/components/datasources/components/new/configure_datasource.tsx (hunk 11 above)
Comment (derek-ho) [reply public/components/datasources/components/new/configure_datasource.tsx:125]: This wouldn't be an issue if we render the form inline, but rendering the form as a child is causing issues

File: opensearch_dashboards.json (hunk 2 above)
Comment (ps48) [reply opensearch_dashboards.json]: removing out should be good here. As the UI is also removed for this as of now.

File: public/components/datasources/components/new/configure_datasource.tsx (hunk 11 above)
Comment (derek-ho) [reply public/components/datasources/components/new/configure_datasource.tsx:125]: Maybe mis-understood your question here - I don't think there is a way to do so, since I need to know the values in the parent in order to send the request and have a way of updating them in the child. Thus I need to pass in value and setter for each of the possible configuration properties. Consolidating it into an object might be cleaner, but I think this  way makes it clear what is going in and as what. Leaning towards keeping it like this for now

File: public/components/datasources/components/manage/data_connection.tsx (hunk 6 above)
Comment (derek-ho) [reply public/components/datasources/components/manage/data_connection.tsx]: Will keep it blank for now, when endpoint is ready will update it separately

File: public/components/datasources/components/manage/data_connection.tsx (hunk 7 above)
Comment (derek-ho) [reply public/components/datasources/components/manage/data_connection.tsx]: Will keep it general for now, and update it later when it is ready

File: public/framework/core_refs.ts (hunk 9 above)
Comment (derek-ho) [reply public/framework/core_refs.ts]: Missed this - was testing out to see about reading feature flag values, but that is not priority right now, thanks for the catch!

File: public/components/datasources/components/manage/connection_configuration.tsx (hunk 10 above)
Comment (derek-ho) [reply public/components/datasources/components/manage/connection_configuration.tsx]: Done

File: public/components/datasources/components/manage/data_connection.tsx [hunk 12]
```diff
@@ -103,6 +108,99 @@ export const DataConnection = (props: any) => {
...
```
[the remark this answers is not in the document]
Comment (derek-ho) [reply public/components/datasources/components/manage/data_connection.tsx]: Good suggestion - made it into a `Default` component, and use it for default and S3 case

**Review by Swiddis — APPROVED** [review]
